File: easy/657-robot-return-to-origin/solution.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        """
        657. Robot Return to Origin
        Time: O(n)
        Space: O(1)
        """
        x = y = 0

        for m in moves:
            if m == 'U':
                y += 1
            elif m == 'D':
                y -= 1
            elif m == 'R':
                x += 1
            else:
                x -= 1

        return x == 0 and y == 0

File: easy/657-robot-return-to-origin/solution.java
/**
 * 657. Robot Return to Origin
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public boolean judgeCircle(String moves) {
        int x = 0, y = 0;

        for (char m : moves.toCharArray()) {
            if (m == 'U') y++;
            else if (m == 'D') y--;
            else if (m == 'R') x++;
            else x--;
        }

        return x == 0 && y == 0;
    }
}
